src/main/java/org/broadinstitute/hellbender/utils/haplotype/AllHaplotypeBAMWriter.java

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.hellbender.utils.haplotype;
+
+import org.broadinstitute.hellbender.utils.GenomeLoc;
+import org.broadinstitute.hellbender.utils.genotyper.ReadLikelihoods;
+import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
+//import org.broadinstitute.hellbender.utils.sam.GATKSAMRecord;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A haplotype bam writer that writes out all haplotypes as reads and then
+ * the alignment of reach read to its best match among the best haplotypes.
+ *
+ * Primarily useful for people working on the HaplotypeCaller method itself
+ */
+class AllHaplotypeBAMWriter extends HaplotypeBAMWriter {
+
+    public AllHaplotypeBAMWriter(final HaplotypeBAMDestination destination) {
+        super(destination);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void writeReadsAlignedToHaplotypes(final Collection<Haplotype> haplotypes,
+                                              final GenomeLoc paddedReferenceLoc,
```

Comment:
If you could eliminate the genome loc and use a Locatable instead, it would be preferable.